waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -1145,19 +1157,31 @@ proc mountRlnRelayDynamic*(node: WakuNode,
   node.wakuRlnRelay = rlnPeer
   return ok(true)
 
-proc readPersistentRlnCredentials*(path: string) : RlnMembershipCredentials {.raises: [Defect, OSError, IOError, Exception].} =
-  info "Rln credentials exist in file"
+proc writeRlnCredentials*(path: string, credentials: RlnMembershipCredentials, password: string) : KfResult[void] {.raises: [Defect, OSError, IOError, Exception].} =
```

Comment:
Here you are using two ways of passing errors:
- Result: `KfResult[void]`
- Exceptions `{.raises: [Defect, OSError, IOError, Exception].}`

We should use a single approach, which should be the `Result[T, R]` for consistency. And convert the exceptions to `err()` by wrapping the code with a `try/except` block.

I understand that this might be out of the scope of this PR, but this is something to consider when evolving the RLN module.

Reply:
Additionally, each module should return the `Result` type of the module. This function, which is exported and is part of the public API of the RLN relay module, returns the result type of the `utils/keyfile` module.

- Either you make this procedure non-public (remove the `*` next to the `proc` name)
- Or return the RLN relay module's `Result` type (e.g. `RlnResult`)